grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/fragment/OutIsaFragment.java

```diff
@@ -60,4 +71,9 @@ public double fragmentCost() {
                                                               Map<Node, Map<Node, Fragment>> edges) {
         return getDirectedEdges(NodeId.NodeType.ISA, nodes, edges);
     }
+
+    @Override
+    public boolean operatesOnEdge() {
+        return true;
```

Comment:
It's always possible for `OutIsaFragment` to start from an edge. We can't tell in advance, so when we build the traversal we must assume we could be starting from an edge.